dev/TreeView/ViewModel.h

```diff
@@ -31,7 +31,9 @@ class ViewModel :
     winrt::event_token NodeCollapsed(const winrt::TypedEventHandler<winrt::TreeViewNode, winrt::IInspectable>& handler);
     void NodeCollapsed(const winrt::event_token token);
     void SelectAll();
-    void ModifySelectByIndex(int index, TreeNodeSelectionState const& state);
+    void SelectItem(winrt::IInspectable const& item);
+    void SelectNode(const winrt::TreeViewNode& node, bool isSelected);
+    void SelectByIndex(int index, TreeNodeSelectionState const& state);
```

Comment:
Why do these all have different ways of specifying state? SelectItem can only put selection state to true, Select node can but the item to true or false and select by index can do any selection state.. seems weird.

Reply:
"Partial selection" is an internal only state. Parent node will be set to partial selected if some of its children are selected, but you can't set an item to partial selected directly. So to the outside world, you can only set selection to true or false, but internally there are actually three states (Selected, PartialSelected, Unselected). But I agree we should probably unify everything to SelectionState to make it less confusing.

Reply:
Should that be part of this PR or a follow up?

Reply:
@StephenLPeters @kaiguo what do you think?

Reply:
Ideally I think all three of these methods would take the form `SelectFoo( Foo item, TreeNodeSelectionState state)`

Reply:
I simplified the API quite a bit now.